### PR TITLE
Feature/gh 10474 read ov data

### LINKED ISSUE #10474: Implement calls to get overview data and normalise to configurator overview model
Implement Adapter which calls CPQ rest service class.
Implement normaliser which takes the result of the calls and maps it to the overview model

### PR DESCRIPTION
**Pull request by Uli-Tiger**
[GH-10474] Read and Normalize data from CPQ for OverviewPage
As we don't have a suitable API, yet,  we call displayConfiguration for every tab to collect all data.

### CODE REVIEW

File: feature-libs/product-configurator/rulebased/cpq/cpq-configurator-overview-normalizer.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    const resultTarget: Configurator.Overview = {
+      ...target,
+      productCode: source.productSystemId,
+      groups: source.tabs?.flatMap((tab) => this.convertTab(tab)),
+      totalNumberOfIssues: this.calculateTotalNumberOfIssues(source),
+    };
+    return resultTarget;
+  }
+
+  convertTab(tab: Cpq.Tab): Configurator.GroupOverview {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/cpq/cpq-configurator-overview-normalizer.ts]: Better make protected and access in test like  `expect(serviceUnderTest['convertTab'](tab).attributes.length).toBe(2);`

File: feature-libs/product-configurator/rulebased/cpq/cpq-configurator-overview-normalizer.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    return {
+      id: tab.id.toString(),
+      groupDescription: tab.name,
+      attributes: tab.attributes?.flatMap((attr) =>
+        this.convertAttribute(attr)
+      ),
+    };
+  }
+
+  convertAttribute(attr: Cpq.Attribute): Configurator.AttributeOverview {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/cpq/cpq-configurator-overview-normalizer.ts]: protected

File: feature-libs/product-configurator/rulebased/cpq/cpq-configurator-rest.service.spec.ts [hunk 3]
```diff
@@ -62,6 +122,11 @@ describe('CpqConfiguratorRestService', () => {
       providers: [
         CpqConfiguratorRestAdapter,
         { provide: OccEndpointsService, useClass: MockOccEndpointsService },
+        {
+          provide: CPQ_CONFIGURATOR_OVERVIEW_NORMALIZER,
+          useClass: DummyOvConverter,
+          multi: true,
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/cpq/cpq-configurator-rest.service.spec.ts]: Not needed here I guess

File: feature-libs/product-configurator/rulebased/cpq/cpq-configurator-rest.service.ts [hunk 4]
```diff
@@ -58,6 +60,71 @@ export class CpqConfiguratorRestService {
...
+              this.callConfigurationDisplay(configId, tab.id.toString())
+            );
+          }
+        });
+
+        // fire requests for remaining tabs and wait until all are finished
+        return forkJoin(tabRequests);
+      }),
+      map(this.mergeTabResults)
+      //tap((resultConfiguration) => console.info(resultConfiguration))
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/cpq/cpq-configurator-rest.service.ts]: Please remove

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: feature-libs/product-configurator/rulebased/cpq/cpq-configurator-overview-normalizer.ts (hunk 1 above)
Comment (Uli-Tiger) [reply feature-libs/product-configurator/rulebased/cpq/cpq-configurator-overview-normalizer.ts]: that was new to me (how to access protected methods in tests)

File: feature-libs/product-configurator/rulebased/cpq/cpq-configurator-rest.service.spec.ts (hunk 3 above)
Comment (Uli-Tiger) [reply feature-libs/product-configurator/rulebased/cpq/cpq-configurator-rest.service.spec.ts]: yes - as i now know how to test protected :-)

**Review by ChristophHi — APPROVED** [review]

**Pull request merged**